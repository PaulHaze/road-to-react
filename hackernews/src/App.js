import React, { Component } from 'react';
// import SearchComp from './components/Search';

// Setup default URL constants
const DEFAULT_QUERY = 'redux';
const PATH_BASE = 'https://hn.algolia.com/api/v1';
const PATH_SEARCH = '/search';
const PARAM_SEARCH = 'query=';

const list = [];

const isSearched = searchTerm => item =>
  item.title.toLowerCase().includes(searchTerm.toLowerCase()) ||
  item.author.toLowerCase().includes(searchTerm.toLowerCase());

class App extends Component {
  // THE CONSTRUCTOR
  constructor(props) {
    super(props);

    this.state = {
      result: null,
      list,
      searchTerm: DEFAULT_QUERY
    };

    this.setSearchTopStories = this.setSearchTopStories.bind(this);
    this.onDismiss = this.onDismiss.bind(this);
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  setSearchTopStories(result) {
    this.setState({ result });
  }

  componentDidMount() {
    const { searchTerm } = this.state;
    fetch(`${PATH_BASE}${PATH_SEARCH}?${PARAM_SEARCH}${searchTerm}`)
      .then(response => response.json())
      .then(result => this.setSearchTopStories(result))
      .catch(error => error);
  }
  // METHODS
  onDismiss(id) {
    const isNotId = item => item.objectID !== id;
    const updatedList = this.state.list.filter(isNotId);
    this.setState({ list: updatedList });
  }

  onSearchChange(event) {
    this.setState({ searchTerm: event.target.value });
  }

  render() {
    /* destructuring the component
           makes searchTerm & list = 
           this.state.searchTerm & this.state.list */
    const { searchTerm, result } = this.state;

    if (!result) {
      return null;
    }

    return (
      <div className="page">
        <div className="interactions">
          <Search value={searchTerm} onChange={this.onSearchChange}>
            <h5>Search</h5>
          </Search>

          <Table
            list={result.hits}
            pattern={searchTerm}
            onDismiss={this.onDismiss}
          />
        </div>

        <div className="interactions">
          <div className="row">
            <div className="col">
              <Button className="" type="button">
                Shop
              </Button>
            </div>
            <div className="col">
              <Button className="" onClick={this.reset} type="button">
                Reload List
              </Button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const Search = ({ value, onChange, children }) => (
  <form>
    {children}
    <input type="text" value={value} onChange={onChange} />
  </form>
);

const Table = ({ list, pattern, onDismiss }) => (
  <div className="table">
    {list.filter(isSearched(pattern)).map(item => (
      <div key={item.objectID} className="table-row">
        <span style={{ width: '30%' }}>
          <a href={item.url}> {item.title},</a>
        </span>
        <span style={{ width: '30%' }}> {item.author}. </span>
        <span style={{ width: '20%' }}> {item.num_comments} comments </span>
        <span style={{ width: '10%' }}> Rating: {item.points} </span>
        <span style={{ width: '10%' }}>
          <Button
            className="button-inline"
            onClick={() => onDismiss(item.objectID)}
          >
            <i className="material-icons medium">cancel</i>
          </Button>
        </span>
        <br />
        <br />
      </div>
    ))}
  </div>
);

const Button = ({ onClick, className = '', children }) => (
  <button onClick={onClick} className={className} type="button">
    {children}
  </button>
);

export default App;
