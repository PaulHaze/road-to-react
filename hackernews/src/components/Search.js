import React from 'react';

export const SearchComp = ({ value, onChange, children }) =>
    <form>
        {children}
        <input
            type="text"
            value={value}
            onChange={onChange} />
    </form>

// export default SearchComp