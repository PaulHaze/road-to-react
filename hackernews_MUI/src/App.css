.App {
  margin-top: 20px;
  background-color: #f5f3f1;
}
