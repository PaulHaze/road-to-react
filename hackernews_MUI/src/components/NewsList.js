import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Icon from '@material-ui/core/Icon';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';

import NewsDetail from './NewsDetail';

const styles = theme => ({
  root: {
    display: 'flex',
    marginTop: theme.spacing(2),
    overflowX: 'hide',
    minWidth: 340,
  },
  table: {
    minWidth: 340,
  },
  tableCell: {
    paddingRight: 3,
    paddingLeft: 8,
  },
});


const NewsList = ({ result, removeStory, classes }) => (
  <Paper className={classes.root}>
    <Table className={classes.table}>
      <TableHead>
        <TableRow>
          <TableCell className={classes.tableCell}>TITLE</TableCell>
          <TableCell className={classes.tableCell} align="left">
            AUTHOR
          </TableCell>
          <TableCell className={classes.tableCell} align="center">
            RATING
          </TableCell>
          <TableCell className={classes.tableCell} align="center">
            <Icon>comment</Icon>
          </TableCell>
          <TableCell className={classes.tableCell} />
        </TableRow>
      </TableHead>
      <TableBody>
        {result
        && result.map(story => (
          <NewsDetail
            key={story.objectID}
            id={story.objectID}
            url={story.url}
            title={story.title}
            author={story.author}
            comments={story.num_comments}
            removeStory={removeStory}
            points={story.points}
          />
        ))}
      </TableBody>
    </Table>
  </Paper>
);

NewsList.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(NewsList);
