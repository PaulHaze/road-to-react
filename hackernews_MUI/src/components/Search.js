import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';

import SearchIcon from '@material-ui/icons/Search';

const useStyles = makeStyles({
  root: {
    padding: '2px 4px',
    display: 'flex',
    alignItems: 'center',
    width: '100%'
  },
  input: {
    marginLeft: 8,
    flex: 1
  },
  iconButton: {
    padding: 10,
    color: '#203b61'
  },
  divider: {
    width: 1,
    height: 28,
    margin: 4
  }
});

export default function Search({ onChange, onSubmit, value }) {
  const classes = useStyles();

  return (
    <Paper className={classes.root}>
      <InputBase
        type="text"
        className={classes.input}
        placeholder="Search HackerNews API"
        value={value}
        onChange={onChange}
        inputProps={{ 'aria-label': 'Search Google Maps' }}
      />
      <Divider className={classes.divider} component={'hr'} />
      <IconButton
        onClick={onSubmit}
        className={classes.iconButton}
        aria-label="Search"
      >
        <SearchIcon />
      </IconButton>
    </Paper>
  );
}
