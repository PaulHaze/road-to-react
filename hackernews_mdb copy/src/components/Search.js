import React, { Component } from 'react';
import { MDBIcon, MDBBtn } from 'mdbreact';

class Search extends Component {
  componentDidMount() {
    if (this.input) {
      this.input.focus();
    }
  }

  render() {
    const { onChange, onSubmit, searchString } = this.props;
    return (
      <form onSubmit={onSubmit} className="input-group md-form form-sm form-1 pl-0">
        <div className="input-group-prepend">
          <span className="input-group-text purple lighten-3" id="basic-text1">
            <MDBIcon icon="" />
          </span>
        </div>
        <input
          onChange={onChange}
          value={searchString}
          className="form-control my-0 py-1"
          type="text"
          placeholder="Search the Hackernews API"
          aria-label="Search"
          ref={(el) => { this.input = el; }}
        />
        <div className="input-group-append">
          <span className="input-group-text purple lighten-3">
            <MDBBtn
              style={{
                maxWidth: '40px',
                maxHeight: '40px',
                minWidth: '30px',
                minHeight: '30px',
                padding: '5px',
                verticalAlign: 'bottom',
              }}
              size="sm"
              color="purple"
              onClick={onSubmit}
              type="submit"
            >
              <MDBIcon className="text-white" icon="search" />
            </MDBBtn>
          </span>
        </div>
      </form>
    );
  }
}


export default Search;
