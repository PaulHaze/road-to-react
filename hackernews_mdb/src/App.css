.App-container {
  border-radius: 8px;
}
