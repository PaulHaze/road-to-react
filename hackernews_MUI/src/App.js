import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import HackerNews from './components/HackerNews';
import './App.css';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(3),
    paddingBottom: '50px',
    backgroundColor: '#f1f1f6',
  },
}));

export default function App() {
  const classes = useStyles();

  return (
    <Paper elevation={15} className={`${classes.root} App `}>
      <h1 className=" text-center">HackerNews API</h1>
      <HackerNews />
    </Paper>
  );
}
