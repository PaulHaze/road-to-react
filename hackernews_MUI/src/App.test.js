/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import Search from './components/Search';
import NewsList from './components/NewsList'

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Search', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Search />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});

describe('Table', () => {
  const props = {
    list: [
      { title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
      { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' },
    ],
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NewsList {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <NewsList {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});