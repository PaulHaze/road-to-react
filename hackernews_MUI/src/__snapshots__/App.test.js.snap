// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation15 makeStyles-root-133 App  MuiPaper-rounded"
>
  <h1
    className=" text-center"
  >
    HackerNews API
  </h1>
  <div
    className="page"
  >
    <div
      className="interactions"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-161 MuiPaper-rounded"
      >
        <div
          className="MuiInputBase-root makeStyles-input-162"
          onClick={[Function]}
        >
          <input
            aria-label="Search Google Maps"
            className="MuiInputBase-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search HackerNews API"
            type="text"
            value="gatsby"
          />
        </div>
        <hr
          className="MuiDivider-root makeStyles-divider-164"
        />
        <button
          aria-label="Search"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-163"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <hr />
      <div
        className="Table-one"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 NewsList-root-207 MuiPaper-rounded"
        >
          <table
            className="MuiTable-root NewsList-table-208"
          >
            <thead
              className="MuiTableHead-root"
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head"
                  scope="col"
                >
                  TITLE
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head MuiTableCell-alignLeft"
                  scope="col"
                >
                  AUTHOR
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head MuiTableCell-alignCenter"
                  scope="col"
                >
                  RATING
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head MuiTableCell-alignCenter"
                  scope="col"
                >
                  <span
                    aria-hidden={true}
                    className="material-icons MuiIcon-root"
                  >
                    comment
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head"
                  scope="col"
                />
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
            />
          </table>
        </div>
        <div
          className="interactions"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root float-right MuiButton-text"
            href="#"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              â†’
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search has a valid snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-161 MuiPaper-rounded"
>
  <div
    className="MuiInputBase-root makeStyles-input-162"
    onClick={[Function]}
  >
    <input
      aria-label="Search Google Maps"
      className="MuiInputBase-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search HackerNews API"
      type="text"
    />
  </div>
  <hr
    className="MuiDivider-root makeStyles-divider-164"
  />
  <button
    aria-label="Search"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-163"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Table has a valid snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 NewsList-root-207 MuiPaper-rounded"
>
  <table
    className="MuiTable-root NewsList-table-208"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head"
          scope="col"
        >
          TITLE
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head MuiTableCell-alignLeft"
          scope="col"
        >
          AUTHOR
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          RATING
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root"
          >
            comment
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root NewsList-tableCell-209 MuiTableCell-head"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    />
  </table>
</div>
`;
